File: src/main/java/io/eliez/fintools/bin2brand/LabeledPrefixRange.java
package io.eliez.fintools.bin2brand;

import lombok.Getter;
import lombok.experimental.Accessors;

@Getter
@Accessors(fluent = true)
class LabeledPrefixRange extends PrefixRange {
    private final String label;

    private LabeledPrefixRange(String label, String lower, String upper) {
        super(lower, upper);
        this.label = label;
    }

    LabeledPrefixRange(String label, PrefixRange range) {
        this(label, range.lower, range.upper);
    }

    String toShortString() {
        return String.format("%s[%s-%s]", label, lower, upper);
    }
}

File: src/main/kotlin/io/eliez/fintools/bin2brand/LabeledPrefixRange.kt
package io.eliez.fintools.bin2brand

internal class LabeledPrefixRange private constructor(val label: String, lower: String, upper: String) : PrefixRange(lower, upper) {

    constructor(label: String, range: PrefixRange) : this(label, range.lower, range.upper) {
    }

    fun toShortString(): String {
        return String.format("%s[%s-%s]", label, lower, upper)
    }
}
